lambda/handler.go

```diff
@@ -15,29 +17,36 @@ type Handler interface {
 	Invoke(ctx context.Context, payload []byte) ([]byte, error)
 }
 
-// lambdaHandler is the generic function type
-type lambdaHandler func(context.Context, []byte) (interface{}, error)
+type handlerOptions struct {
+	Handler
+	baseContext context.Context
+	escapeHTML  bool
```

Comment:
If this is only relevant to response serialization, maybe we can make it explicit in the property name (eg. `escapeResponseHTML` or something along those lines)

Reply:
I agree. I would even go a bit further and prefix/suffix these with something like `ResponseJSON` so it would be something like:

```
responseJSONEscapeHTML
responseJSONPrefix
responseJSONIndex
```

Alternatively you could have customers pass in a `newJSONEncoder` function which they could configure however they like (or use a different JSON library). It would have the following signature:

```
newJSONEncoder func(out io.Writer) error
```

That might be the most flexible. Could even throw in a `newJSONDecoder` so users could have complete control over the JSON part of the process.

Reply:
I did play around with the custom encoder idea before pushing this PR! However didn't come up with something that still elegantly worked with the stdlib json NewEncoder. My failed ideas looked something like this: https://play.golang.com/p/To4q1sKBbjm 

Something like this would work:
```
func WithEncoder(encoderFactory func(io.Writer) func(any) error) Option {
  return Option(func(h *handlerOption) {
    h.encoderFactory = encoderFactory
  })
}
// use example with stdlib json
encoderOption := WithEncoder(func(w io.Writer) func(any) error {
  return json.NewEncoder(w).Encode
})
```
But i'd want to mull on it a little more, and look at other popular encoders, to see if there's not something more ergonomic that can be figured out

Reply:
yea, I personally like the `WithEncoder` pattern over having three new json-specific variables but either way ultimately works.